crates/bevy_text/src/input.rs

```diff
@@ -0,0 +1,1312 @@
+use core::time::Duration;
+
+use crate::buffer_dimensions;
+use crate::load_font_to_fontdb;
+use crate::CosmicFontSystem;
+use crate::Font;
+use crate::FontAtlasSets;
+use crate::FontSmoothing;
+use crate::Justify;
+use crate::LineBreak;
+use crate::LineHeight;
+use crate::PositionedGlyph;
+use crate::TextBounds;
+use crate::TextError;
+use crate::TextFont;
+use crate::TextLayoutInfo;
+use crate::TextPipeline;
+use alloc::collections::VecDeque;
+use bevy_asset::Assets;
+use bevy_asset::Handle;
+use bevy_derive::Deref;
+use bevy_derive::DerefMut;
+use bevy_ecs::change_detection::DetectChanges;
+use bevy_ecs::change_detection::DetectChangesMut;
+use bevy_ecs::component::Component;
+use bevy_ecs::entity::Entity;
+use bevy_ecs::event::EntityEvent;
+use bevy_ecs::hierarchy::ChildOf;
+use bevy_ecs::lifecycle::HookContext;
+use bevy_ecs::prelude::ReflectComponent;
+use bevy_ecs::query::Changed;
+use bevy_ecs::query::Or;
+use bevy_ecs::resource::Resource;
+use bevy_ecs::schedule::SystemSet;
+use bevy_ecs::system::Commands;
+use bevy_ecs::system::Query;
+use bevy_ecs::system::Res;
+use bevy_ecs::system::ResMut;
+use bevy_ecs::world::DeferredWorld;
+use bevy_ecs::world::Ref;
+use bevy_image::Image;
+use bevy_image::TextureAtlasLayout;
+use bevy_math::IVec2;
+use bevy_math::Rect;
+use bevy_math::UVec2;
+use bevy_math::Vec2;
+use bevy_reflect::prelude::ReflectDefault;
+use bevy_reflect::Reflect;
+use bevy_time::Time;
+use cosmic_text::Action;
+use cosmic_text::BorrowedWithFontSystem;
+use cosmic_text::Buffer;
+use cosmic_text::BufferLine;
+use cosmic_text::Edit;
+use cosmic_text::Editor;
+use cosmic_text::Metrics;
+pub use cosmic_text::Motion;
+use cosmic_text::Selection;
+/// Systems handling text input update and layout
+#[derive(Debug, Hash, PartialEq, Eq, Clone, SystemSet)]
+pub struct TextInputSystems;
+
+/// Basic clipboard implementation that only works within the bevy app.
+#[derive(Resource, Default)]
+pub struct Clipboard(pub String);
```

Comment:
I am tempted to cut this from the initial PR: we can make this simpler and improve reviewability / controversy.

Reply:
Actually, the implementation looks very simple and uncontroversial. This can stay.